packages/hash/frontend/src/components/BlockLoader/data-map-editor.tsx

```diff
@@ -272,28 +283,24 @@ export const DataMapEditor = ({
                   {description ?? "No description provided"}
                 </Typography>
               </Box>
-              <Box width="30%">
+              <Box flex={1}>
                 <Select
                   size="small"
                   defaultValue={
-                    schemaMap?.transformations?.[targetPath]?.sourceKey ??
+                    schemaMap.transformations?.[targetPath]?.sourceKey ??
                     "__nothing"
                   }
                   onChange={(evt) =>
-                    updateSchemaMap((existingMap) => ({
-                      ...(existingMap ?? { mapId }),
-                      transformations: {
-                        ...existingMap.transformations,
-                        [targetPath]: {
-                          default:
-                            existingMap.transformations?.[targetPath]?.default,
-                          sourceKey:
-                            evt.target.value === "__nothing"
-                              ? undefined
-                              : evt.target.value,
-                        },
-                      },
-                    }))
+                    onSchemaMapChange((existingMap) =>
+                      produce(existingMap, (draftMap) => {
+                        draftMap.transformations ??= {};
+                        draftMap.transformations[targetPath] ??= {};
+                        draftMap.transformations[targetPath]!.sourceKey =
+                          evt.target.value === "__nothing"
+                            ? undefined
+                            : evt.target.value;
+                      }),
+                    )
```

Comment:
Immer has a great feature where when you only pass it a function, it returns a function, which means you can do this

```suggestion
                    onSchemaMapChange(produce((draftMap) => {
                      draftMap.transformations ??= {};
                      draftMap.transformations[targetPath] ??= {};
                      draftMap.transformations[targetPath]!.sourceKey =
                        evt.target.value === "__nothing"
                          ? undefined
                          : evt.target.value;
                    }))
```